pkg/kubectl/cmd/util/helpers.go

```diff
@@ -83,6 +83,58 @@ func checkErr(err error, handleErr func(string)) {
 	handleErr(msg)
 }
 
+// CheckCustomErr is like CheckErr except a custom prefix error
+// string may be provied to help produce more specific error messages.
+// For example, for the update failed case this function could be called
+// with:
+//    cmdutil.CheckCustomErr("Update failed", err)
+// This function supresses the detailed output that is produced by CheckErr
+// and specifically the field is erased and the error message has the details
+// of the spec removed. Unfortunately, what starts off as a detail message is
+// a sperate field ends up being concatentated into one string which contains
+// the spec and the detail string. To avoid significant refactoring of the error
+// data structures we just extract the required detail string by looking for it
+// after "}': " which is horrible but expedient.
```

Comment:
TODO: We should create an error struct with information embedded in fields, not in a string.
(or touch on how this should be fixed up after v1)